sympy/polys/polyroots.py

```diff
@@ -861,6 +862,31 @@ def roots(f, *gens,
     >>> roots([1, 0, -1])
     {-1: 1, 1: 1}
 
+    ``roots`` will only return roots expressible in radicals.
+    If the given polynomial has some or all of its roots inexpressible in radicals,
+    the result of ``roots`` will be incomplete or empty respectively.
+
+    Example where result is incomplete:
+
+    >>> roots((x-1)*(x**5-x+1), x)
+    {1: 1}
+
+    Example where result is empty:
+
+    >>> roots(x**7-3*x**2+1, x)
+    {}
+
+    The result produced by ``roots`` is complete if and only
+    if the sum of the multiplicity of each root is equal to the
+    degree of the polynomial. This can be checked as follows:
+
+    >>> f = x**3-2*x**2+1
+    >>> sum(roots(f, x).values()) == degree(f, x)
+    True
```

Comment:
I wonder whether it might also be good to have a `strict` argument for `roots`...

Reply:
I have tried adding a `strict` flag to `roots`. It simplifies the code in `solve_generic`, and does not change any existing behavior, so I think this is a good idea.